From 18bb69c5dca472146d1b93812366efe5709ed1d6 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:12 -0400
Subject: [scsi] hpsa: Acknowledge controller events in ioaccell mode 2 as well as mode 1

Message-id: <1410186734-19986-22-git-send-email-thenzl@redhat.com>
Patchwork-id: 92035
O-Subject: [RHEL6.6 e-stor PATCH 21/83] hpsa: Acknowledge controller events in ioaccell mode 2 as well as mode 1
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 1f7cee8c7d44ad360a16c7e87831b4568419322b)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index aee8cf9..6ffaff7 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -5523,7 +5523,8 @@ static int hpsa_kickoff_rescan(struct ctlr_info *h)
  char *event_type;
 
  /* Ask the controller to clear the events we're handling. */
- if (h->transMethod & (CFGTBL_Trans_io_accel1) &&
+ if ((h->transMethod & (CFGTBL_Trans_io_accel1
+   | CFGTBL_Trans_io_accel2)) &&
   (h->events & HPSA_EVENT_NOTIFY_ACCEL_IO_PATH_STATE_CHANGE ||
    h->events & HPSA_EVENT_NOTIFY_ACCEL_IO_PATH_CONFIG_CHANGE)) {
 
diff --git a/drivers/scsi/hpsa_cmd.h b/drivers/scsi/hpsa_cmd.h
index 9e00988..6b336e8 100644
--- a/drivers/scsi/hpsa_cmd.h
+++ b/drivers/scsi/hpsa_cmd.h
@@ -133,6 +133,7 @@
 #define CFGTBL_Trans_Simple     0x00000002l
 #define CFGTBL_Trans_Performant 0x00000004l
 #define CFGTBL_Trans_io_accel1 0x00000080l
+#define CFGTBL_Trans_io_accel2 0x00000100l
 #define CFGTBL_Trans_use_short_tags 0x20000000l
 #define CFGTBL_Trans_enable_directed_msix (1 << 30)
 
-- 
1.7.1